pkg/tsdb/elasticsearch/response_parser.go

```diff
@@ -825,7 +825,9 @@ func nameFrames(queryResult backend.DataResponse, target *Query) {
 			// another is "number"
 			valueField := frame.Fields[1]
 			fieldName := getFieldName(*valueField, target, metricTypeCount)
-			frame.Name = fieldName
+			if fieldName != "" {
+				valueField.SetConfig(&data.FieldConfig{DisplayNameFromDS: fieldName})
```

Comment:
nitpick: is there a chance that `valueField.Config` already exists with some other values at this point? if it is possible, here we are removing those values, and we should do something like:
```go
if fieldName.Config == nil {
    fieldName.Config = data.FieldConfig{}
}
```

but if we know it's empty, we can of course go with the simpler approach.

Reply:
Fixed in https://github.com/grafana/grafana/pull/66039/commits/0f5a24e92e4e938b1423f46b38431a51f0324f12